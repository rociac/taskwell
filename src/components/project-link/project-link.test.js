import ProjectLink from './project-link.component';
import renderer from 'react-test-renderer';
import { cleanup } from '@testing-library/react';
import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';

afterEach(cleanup);

describe('Project Link', () => {
  it('renders properly with props', () => {
    const history = createMemoryHistory();
    const mockProps = {
      to: 'www.example.com',
      name: 'Test Project',
      image: 'www.example.com/image',
      type: 'Software'
    };
    const tree = renderer.create(<Router history={history}><ProjectLink {...mockProps} /></Router>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
