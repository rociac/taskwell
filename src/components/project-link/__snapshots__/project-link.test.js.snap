// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Project Link renders properly with props 1`] = `
<div
  className="container"
>
  <div
    className="imageContainer"
  >
    <img
      alt="project-preview"
      className="image"
      src="www.example.com/image"
    />
  </div>
  <div
    className="textContainer"
  >
    <a
      className="link"
      href="/www.example.com"
      onClick={[Function]}
    >
      <p
        className="title"
      >
        Test Project
      </p>
    </a>
    <p
      className="type"
    >
      Software
    </p>
  </div>
</div>
`;
